js/components/Drives.tsx

```diff
@@ -1,26 +1,53 @@
-import { h, Fragment } from 'preact';
-import {useEffect, useState } from 'preact/hooks';
+import { h } from 'preact';
+import { useCallback, useEffect, useState } from 'preact/hooks';
 import Disk2, { Callbacks } from '../cards/disk2';
 import Apple2IO from '../apple2io';
 import { DiskII, DiskIIData } from './DiskII';
+import SmartPort from 'js/cards/smartport';
+import CPU6502 from 'js/cpu6502';
+import { BlockDisk } from './BlockDisk';
+import { ErrorModal } from './ErrorModal';
+import { ProgressModal } from './ProgressModal';
+import { loadHttpUnknownFile, getHashParts, loadJSON } from './util/files';
+import { useHash } from './hooks/useHash';
+import { DriveNumber } from 'js/formats/types';
+import { Ready } from './util/promises';
+
+import styles from './css/Drives.module.css';
 
 /**
  * Interface for Drives component.
  */
 export interface DrivesProps {
+    cpu: CPU6502 | undefined;
     io: Apple2IO | undefined;
+    enhanced: boolean;
     sectors: number;
+    ready: Ready;
 }
 
 /**
  * Drive interface component. Presents the interface to load disks.
- * Provides the callback to the Disk2 object to update the DiskII
- * components.
+ * Provides the callback to the Disk2 and SmartPort objects to update
+ * the DiskII and BlockDisk components.
+ * Handles initial loading of disks specified in the hash.
  *
+ * @cpu CPU object
  * @param io Apple I/O object
+ * @param sectors 13 or 16 sector rom mode
+ * @enhanced Whether to create a SmartPort ROM device
+ * @ready Signal disk availability
  * @returns Drives component
  */
-export const Drives = ({ io, sectors }: DrivesProps) => {
+export const Drives = ({ cpu, io, sectors, enhanced, ready }: DrivesProps) => {
+    const [current, setCurrent] = useState(0);
+    const [error, setError] = useState<unknown>();
+    const [total, setTotal] = useState(0);
```

Comment:
Perhaps group `current` and `total` together? (I suppose a single progress object could also work.)

Reply:
I think I will try to clean this up when I address the multiple loading issue.